lib/rdkafka/consumer.rb

```diff
@@ -441,5 +441,26 @@ def each
         end
       end
     end
+
+    def each_batch(max_items: 100, max_latency_ms: 250, &block)
+      slice = []
```

Comment:
Maybe there's a way to not have to create a new array?

Reply:
> Maybe there's a way to not have to create a new array?

I believe it's essential for correct behavior to `dup` the array that we yield; otherwise, clients could accidentally muck with our array even as we're filling it back up, or create other unintended behaviours by accident. However I was able to convert the `slice = []` assignment on line 461 to `clear`, for what it's worth.